### PR TITLE
CELDEV-1173 - add a imgSrcParamsattribute to cel-data-img 

### PR DESCRIPTION
**Pull request by fpichler**
https://synjira.atlassian.net/browse/CELDEV-1173

### CODE REVIEW

File: src/main/webapp/resources/celDynJS/celData/cel-data.mjs [hunk 1]
```diff
@@ -145,6 +133,19 @@ export class CelDataImage extends CelData {
...
 
+  get additionalParams() {
+    return this.getAttribute('additional-params') ?? '';
+  }
+
+  urlImageSrc(data) {
+    const src = data?.[this.field];
+    if (src) {
+      const del = (src.indexOf('?') > -1) ? '&' : '?';
+      return src + del + this.additionalParams;
```
Comment (msladek) [inline src/main/webapp/resources/celDynJS/celData/cel-data.mjs]: use URL/URLSearchParams:
```suggestion
      const url = new URL(src);
      for (const [key, value] of new URLSearchParams(this.additionalParams)) {
        url.searchParams.append(key, value);
      }
      return url.toString();
```

File: src/main/webapp/resources/celDynJS/celData/cel-data.mjs [hunk 2]
```diff
@@ -145,6 +133,19 @@ export class CelDataImage extends CelData {
     return this.getAttribute('loading') ?? '';
   }
 
+  get additionalParams() {
+    return this.getAttribute('additional-params') ?? '';
+  }
+
+  urlImageSrc(data) {
```
Comment (msladek) [inline src/main/webapp/resources/celDynJS/celData/cel-data.mjs:140]: is there a need for this method to be public?
```suggestion
  #urlImageSrc(data) {
```

File: src/main/webapp/resources/celDynJS/celData/cel-data.mjs [hunk 3]
```diff
@@ -145,6 +133,19 @@ export class CelDataImage extends CelData {
     return this.getAttribute('loading') ?? '';
   }
 
+  get additionalParams() {
+    return this.getAttribute('additional-params') ?? '';
```
Comment (msladek) [inline src/main/webapp/resources/celDynJS/celData/cel-data.mjs]: for clarity I'd prefer this attribute to be named `additional-src-params` or similar

**Review by msladek — CHANGES_REQUESTED** [review]

File: src/main/webapp/resources/celDynJS/celData/cel-data.mjs (hunk 2 above)
Comment (fpichler) [reply src/main/webapp/resources/celDynJS/celData/cel-data.mjs:140]: yes, only public methods can be called in subclasses. Therefor it is by design that the CelDataImage is extensible.

**Review by msladek — APPROVED** [review]

**Pull request merged**